lib/livebook_web/live/hub_live/fly_component.ex

```diff
@@ -96,88 +93,120 @@ defmodule LivebookWeb.HubLive.FlyComponent do
                   >
                     <.remix_icon icon="refresh-line" class="text-xl" />
                   </button>
+                  <%= error_tag(f, :hub_color) %>
                 </div>
               </div>
             </div>
           </div>
 
-          <%= submit("Save", class: "button-base button-blue", phx_disable_with: "Saving...") %>
+          <%= submit("Save",
+            class: "button-base button-blue",
+            phx_disable_with: "Saving...",
+            disabled: not @valid?
+          ) %>
         <% end %>
       </.form>
     </div>
     """
   end
 
   defp load_data(%{assigns: %{operation: :new}} = socket) do
-    assign(socket, data: %{}, select_options: [], apps: [])
+    assign(socket,
+      changeset: Fly.change_hub(%Fly{}),
+      selected_app: nil,
+      select_options: [],
+      apps: [],
+      valid?: false
+    )
   end
 
   defp load_data(%{assigns: %{operation: :edit, hub: fly}} = socket) do
-    data = %{
-      "access_token" => fly.access_token,
-      "application_id" => fly.application_id,
-      "hub_name" => fly.hub_name,
-      "hub_color" => fly.hub_color
-    }
-
     {:ok, apps} = FlyClient.fetch_apps(fly.access_token)
-    opts = select_options(apps, fly.application_id)
+    params = Map.from_struct(fly)
 
-    assign(socket, data: data, selected_app: fly, select_options: opts, apps: apps)
+    assign(socket,
+      changeset: Fly.change_hub(fly, params),
```

Comment:
Can't we just do `Fly.change_hub(hub, %{})`?

Reply:
Without using `Fly.change_hub(hub, params)` I can't get all the params from event `validate` and it fails on event `save`, because it doesn't change the password and the select inputs